README.md

```diff
@@ -37,123 +41,50 @@ registry := pkg.NewProtocolRegistry()
 
 ### Registry new Protocol Operation
 
-To register a new protocol operation, you can use the `RegisterProtocolOperation` function:
+To register a new protocol operation, you can use the
+`RegisterProtocolOperation` function:
 
 ```go
-registry.RegisterProtocolOperation(protocol.Name, protocol.Action, protocol.ChainID, &pkg.GenericProtocolOperation{
-    DynamicOperation: pkg.DynamicOperation{
-        Protocol: protocol.Name,
-        Action:   pkg.SupplyAction,
-        ChainID:  big.NewInt(1),
-    },
-})
+registry.RegisterProtocolOperation(protocol.Address, protocol.ChainID, implementationStruct)
 ```
 
 ### Retrieving Protocol Operations
 
 To retrieve a protocol operation, you can use the `GetProtocolOperation` function:
 
 ```go
-operation, err := registry.GetProtocolOperation(pkg.AaveV3, pkg.SupplyAction, big.NewInt(1))
+operation, err := registry.GetProtocolOperation(protocol.Address, big.NewInt(1))
 if err != nil {
     // Handle the error
 }
 ```
 
 ### Generating Calldata
 
-To generate calldata for a specific operation, you can use the `GenerateCalldata` method of the retrieved operation:
+To generate calldata for a specific operation, you can use the
+`GenerateCalldata` method of the retrieved operation:
 
 ```go
-args := []interface{}{
-    common.HexToAddress("0x1f9840a85d5aF5bf1D1762F925BDADdC4201F984"),
-    big.NewInt(1000000000000000000),
-    common.HexToAddress("0x0000000000000000000000000000000000000000"),
-    uint16(10),
-   }
-calldata, err := operation.GenerateCalldata(pkg.LoanKind, args)
+calldata, err := operation.GenerateCalldata(pkg.LoanKind, GenerateCalldataOptions{
+  Asset: common.HexToAddress("0x1f9840a85d5aF5bf1D1762F925BDADdC4201F984"),
+  Recipient: common.HexToAddress("0xB4FBF271143F4FBf7B91A5ded31805e42b2208d6"),
+  Amount:    big.NewInt(1e18),
+})
 if err != nil {
     // Handle the error
 }
 ```
 
-## Adding New Protocols and Operations
-
-To add support for a new protocol and its operations, follow these steps:
```

Comment:
Looks like the section is removed without a replacement. What are the steps to add support for a new Protocol with this refactoring?